spec/requests/dfe_sign_in_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe 'DfE Sign-in' do
+  before do
+    ensure_omniauth_has_not_been_mocked_in_another_test
+  end
+
+  context 'when DfE Sign-in respond with an OIDC payload for authentication purposes' do
+    context 'when that payload is unauthorised' do
+      it 'redirects the user to the not authorised page' do
+        params = {
+          'code' => 'a-long-secret',
+          'session_state' => '123.456'
+        }
+
+        get '/auth/dfe/callback', params: params
+
+        expect(response.status).to eq(302)
+        expect(response.body).to eq('Redirecting to /401...')
+      end
+    end
+  end
+
+  context 'when a user is linked back to our service with a redirect' do
+    it 'routes the request back to the new sign-in page' do
+      request = get '/auth/dfe/callback'
```

Comment:
can we use the `auth_dfe_callback_path` path helpers here - as it's easier to spot failures if anything changes.
I can't recall if request specs have access to path helpers. Same for line 16.

Reply:
It's hard coded in the routes as there is no resource to attach the routing to so I don't think this can be done.